### PR TITLE
chore(external docs): Add `remove_fields` transform metadata

### PR DESCRIPTION
**Pull request by binarylogic**
Completes https://github.com/timberio/vector/issues/4303 for `remove_fields`

### CODE REVIEW

File: docs/reference/components/transforms/remove_fields.cue [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  }
+
+  support: {
+      input_types: ["log"]
+
+    platforms: {
+      "aarch64-unknown-linux-gnu": true
+      "aarch64-unknown-linux-musl": true
+      "x86_64-apple-darwin": true
+      "x86_64-pc-windows-msv": true
```
Comment (Hoverbear) [inline docs/reference/components/transforms/remove_fields.cue:30]: Noting we discussed fixing this later.

**Review by Hoverbear — APPROVED** [review]

**Comment by binarylogic** [comment]
@Hoverbear could you do the steps outlined in #4303? Let me know if you have questions.

**Comment by Hoverbear** [comment]
Oh, sure.

**Review by Hoverbear — APPROVED** [review]

**Pull request merged**